server/grpc/service.go

```diff
@@ -26,3 +27,20 @@ func (s *ServiceServer) Create(ctx context.Context, request *protobuf_api.Create
 	}
 	return &protobuf_api.CreateServiceResponse{Sid: srv.Sid, Hash: srv.Hash}, nil
 }
+
+// Delete deletes service by hash or sid.
+func (s *ServiceServer) Delete(ctx context.Context, request *protobuf_api.DeleteServiceRequest) (*protobuf_api.DeleteServiceResponse, error) {
+	srv, err := s.sdk.GetService(request.HashOrSid)
+	if err != nil {
+		return nil, err
+	}
+	// first, check if service has any running instances.
```

Comment:
not a fan of having this check here. high level sdk pkg can be a good place to put the service deletion logic. because it has access to both instancesdk and servicesdk. with this improvements api will not have any extra logic which is a good thing if there'll be other type of apis in future